src/transaction.ts

```diff
@@ -300,7 +327,15 @@ class Transaction extends DatastoreRequest {
    *   });
    * });
    */
-  createQuery(namespace: string, kind?: string | string[]): Query {
+  createQuery(
+    namespaceOrKind?: string | string[],
+    kind?: string | string[]
+  ): Query {
+    let namespace = namespaceOrKind as string;
+    if (arguments.length < 2) {
+      kind = namespaceOrKind;
+      namespace = this.namespace!;
```

Comment:
How does this fix #486 exactly?

`this.namespace` is just an [alias to `this.datastore.namespace](https://github.com/googleapis/nodejs-datastore/blob/master/src/transaction.ts#L75)` and the [`datastore.createQuery`](https://github.com/googleapis/nodejs-datastore/blob/master/src/index.ts#L647) method contains the same logic here.

Reply:
@callmehiphop  fair question,
while passing only kind in `transaction.createQuery` it internally calls `datastore.createQuery` with two arguments namespace(that user pass as kind) and kind as `undefined`.
in above scenario `datastore.createQuery` argument condition will not fulfill(as it got two arguments) so it will take first argument as namespace and second as kind that is not correct. 

it is necessary to check argument condition in `transaction.createQuery` as well.

Reply:
Oh, so would you say that the check in `datastore.createQuery` is bad? And instead of checking for the number of arguments, we should be checking if the second argument is not undefined?

Reply:
true @callmehiphop, made changes in `datastore.createQuery` condition as per your suggestion. keeping overload for `transaction.createQuery` now.